app/controllers/admin/data_sources_controller.rb

```diff
@@ -7,12 +7,18 @@ def show
       respond_to do |format|
         format.html
         format.csv do
-          send_data @data_source.meters.to_csv,
+          send_data @data_source.to_csv,
           filename: "#{t('common.application')}-#{@data_source.name}-meters-#{Time.zone.now.iso8601}".parameterize + '.csv'
         end
       end
     end
 
+    def deliver
+      @data_source = DataSource.find(params[:data_source_id])
+      SendDataSourceReportJob.perform_now(to: current_user.email, data_source_id: @data_source.id)
```

Comment:
This should be `perform_later` otherwise is runs the job immediately. We want it to run in the background within GoodJobs

Reply:
👍 done